### DISCUSSION TITLE
Feature request: implement new upstream realtime database get() method

### DISCUSSION THREAD

**Discussion by cbarrios** [invertase/react-native-firebase#4933] [Ideas]
Hey, it would be of great help if we can get the get() method to work for the realtime database. This is a new feature added recently by the Firebase team. This way we get data once by looking at the server, which is extremely useful when persistence is enabled since we don't need to hack our way to use on/off methods.

Thank you.

**Comment by mikehardy** [comment]
Hi there!

Do you mean this ? 
- JS API that we would mirror: https://firebase.google.com/support/release-notes/js#version_810_-_november_19_2020
- iOS implementation release: https://firebase.google.com/support/release-notes/ios#realtime-database_1
- Android implementaiton release: https://firebase.google.com/support/release-notes/android#realtime-database_v19-6-0

Certainly looks possible, we could certainly merge any PR that implemented it and given that this module is pretty mature (and pretty thoroughly tested!) there is probably a great deal of existing implementation of similar APIs to pattern off of if you wanted to propose a PR?

The [README for the e2e tests "app"](https://github.com/invertase/react-native-firebase/blob/master/tests/README.md) that we use for feature development and verification is very thorough and I have found it to be a very efficient development process if you want to give it a try - usually a matter of adding something to (in this case) database/lib/index.js (and index.d.ts) to call native, add a quick stub in the native code (just to have something for it to call), add a new test to the appropriate `database/e2e/**/*e2e.js` file (with it set as `testname.only` to start) and poof - you have new code executing with very fast code+test turnarounds. Then just find the most similar API, copy it over and bend it to fit this new underlying API. Should work?

**Reply by cbarrios** [reply]
Yes that feature is the one I'm referring to. It recently got fixed for the JS SDK for some issue when filtering. I just wanted to let you guys know so that it is added since it is of great value. Right now I am using on()/off() to achieve the same result but having get() would make that much cleaner. Thank you!
